### ISSUE TITLE
Merge Function - Multiple Copies of same PDf merged - Page Numbering

### ISSUE THREAD

**Issue by Stitchard989** [QuestPDF/QuestPDF#1081]
**Describe the bug**
I have created a new QuestPDF document, I then want 3 copies of this  PDF merged into 1 merged document. I have a for loop to merge according to how many copies are required, using  .UseOriginalPageNumbers()
If I have a Quantity of 1 copy no problems, same if I have 2 copies merged no problem , if I request 3 or more the  .UseOriginalPageNumbers() does not seem to work correctly.
If I try 2 copies , Page1 prints  "Page 1 of 1" , Page2 prints "Page 1 of 1" - Which is correct
If I try 3 copies , Page1 now prints  "Page 1 of 2" , Page2 prints "Page 2 of 2" and Page3 prints "Page 1 of 1"
If I try 4 copies , Page1 now prints  "Page 1 of 3" , Page2 prints "Page 2 of 3" , Page3 prints "Page 3 of 3" and Page4 prints "Page 1 of 1"
I hope that makes sense

**To Reproduce**
 int printQty = _context.PaperworkSettings.AutoPrintQuantity; 
 var document = new DeliveryDocument(delivery);
 var mergedDocument = Document.Merge(document);
 
   for (int i = 1; i < printQty; i++)
   {
     // Re-merge the same document again
     mergedDocument = Document.Merge(mergedDocument, document)
         .UseOriginalPageNumbers();  // Keep original page numbers
   }  

**Expected behavior**
2 copies , Page1 should print  "Page 1 of 1" , Page2 should print "Page 1 of 1" - Which is currently working
3 copies , Page1 should print  "Page 1 of 1" , Page2 should print "Page 1 of 1" and Page3 should print "Page 1 of 1"
4 copies , Page1 should print  "Page 1 of 1" , Page2 should print "Page 1 of 1" , Page3 should print "Page 1 of 1" and Page4 should print "Page 1 of 1"

**Environment**
C# - .NET8 - Visual studio 2022
QuestPDF Version 2024.12.1
Ooperating system Windows x64

**Comment by MarcinZiabek** [comment]
I have investigated the issue. Please avoid nesting the merging operation. If you need to merge multiple copies, provide the entire collection and execute the Document.Merge() method only once.

To prevent unintended behavior, I will enhance the library to throw an exception if documents are merged in a nested or recursive manner.

https://www.questpdf.com/concepts/merging-documents.html

```c#
Document
    .Merge(
        GenerateReport("Document 1", 5),
        GenerateReport("Document 2", 10),
        GenerateReport("Document 3", 15),
        GenerateReport("Document 4", 20),
        GenerateReport("Document 5", 25))
    .UseOriginalPageNumbers()
    .GeneratePdf("merged-original-page-numbers.pdf");
```

I’m closing this issue. If you experience a similar problem or have a related request, feel free to open a new GitHub issue. I appreciate your feedback! ❤️

**Comment by Stitchard989** [comment]
That's great , thank you for the explanation , makes sense. 
Much appreciated
